File: ontrack-extension-general/src/main/java/net/nemerosa/ontrack/extension/general/LinkProperty.java
package net.nemerosa.ontrack.extension.general;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import net.nemerosa.ontrack.model.support.NameValue;

import java.util.Arrays;
import java.util.List;

@Data
@AllArgsConstructor(access = AccessLevel.PUBLIC)
public class LinkProperty {

    private final List<NameValue> links;

    public static LinkProperty of(String name, String value) {
        return new LinkProperty(
                Arrays.asList(
                        new NameValue(name, value)
                )
        );
    }
}

File: ontrack-extension-general/src/main/java/net/nemerosa/ontrack/extension/general/LinkProperty.kt
package net.nemerosa.ontrack.extension.general

import net.nemerosa.ontrack.model.support.NameValue

data class LinkProperty(
        val links: List<NameValue>
) {

    companion object {
        @JvmStatic
        fun of(name: String, value: String): LinkProperty {
            return LinkProperty(
                    listOf(
                            NameValue(name, value)
                    )
            )
        }

        fun of(vararg entries: Pair<String, String>): LinkProperty =
                LinkProperty(
                        entries.map {
                            NameValue(it.first, it.second)
                        }
                )
    }
}
